plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs

```diff
@@ -49,35 +48,26 @@ of 'stringToBuiltinString' (since the newtype constructor vanishes), so we have
 some obfuscation to the body to prevent it inlining.
 -}
 
-obfuscatedId :: a -> a
-obfuscatedId a = a
-{-# NOINLINE obfuscatedId #-}
-
 stringToBuiltinByteString :: Haskell.String -> BuiltinByteString
 stringToBuiltinByteString str = encodeUtf8 $ stringToBuiltinString str
-{-# INLINABLE stringToBuiltinByteString #-}
+{-# NOINLINE stringToBuiltinByteString #-}
 
 stringToBuiltinString :: Haskell.String -> BuiltinString
--- To explain why the obfuscatedId is here
--- See Note [noinline hack]
-stringToBuiltinString str = obfuscatedId (BuiltinString $ Text.pack str)
-{-# INLINABLE stringToBuiltinString #-}
+stringToBuiltinString str = BuiltinString (Text.pack str)
+{-# NOINLINE stringToBuiltinString #-}
```

Comment:
We [should](https://github.com/IntersectMBO/plutus/issues/6269) use `OPAQUE`, not `NOINLINE`.